simplesat/dependency_solver.py

```diff
@@ -52,12 +52,7 @@ def _create_rules(self, request):
             pool.package_id(package)
             for package in pool.what_provides(requirement)
         ]
-        self._policy.add_packages_by_id(requirement_ids)
-
-        # Add installed packages.
```

Comment:
`add_requirements` now implies hard requirements, not just packages we should prefer if possible. The `Policy` gets a reference to `installed_repository` and can figure out on its own which packages are currently installed if it wants to.